### DISCUSSION TITLE
How to compile and run ABS in Java

### DISCUSSION THREAD

**Discussion by aulia-adil** [abstools/abstools#304] [Q&A] [answered]
So I read the ABS manual in [here](https://abs-models.org/manual/#-abs-backends) but I didn't find a way to compile ABS file with Java. I only found the manual for Erlang, Haskell, and Maude. I have tried something like `absc -j file.abs` and then I got the `gen` directory. But I didn't know how to run that `gen` directory. So my question is how to compile and run ABS in Java?

**Comment by rudi** [comment, marked as answer]
Hi, something like `java -cp gen:absfrontend.jar Module.Main` should work - I’m currently traveling and can’t check properly so I looked in the sources of the emacs mode. 

Be advised that the Java backend hasn’t been actively developed in some years and misses features like the model api and time model. Patches are, as they say, highly welcome :)

**Reply by rudi** [reply]
The same holds even more for the Haskell backend, and I never got the Scala backend to run, even though it might be the most promising one to run on the JVM …

**Reply by aulia-adil** [reply]
Thank you for your answer @rudi. I asked this because I have this old ABS project and it uses Java as the backend. I guess I have to tell my supervisor that Java ABS is deprecated and it's time to move to Erlang.

**Reply by rudi** [reply]
Cool - let me know how it goes, and come back with any questions. Apologies in advance for late answers since I’m traveling this week

**Comment by vlad-serbanescu** [comment]
Hi, I tried catching up with the main branch changes for the Scala backend, but never had the time to complete it. If there is interest in it and someone who can spend significant time on it, maybe I can meet with them and possibly work together to get it up to speed?

**Reply by rudi** [reply]
Hi Vlad, good to hear from you! I think starting from your branch would be ok - what I need mostly is a sequence of shell commands that take me from an abs source file containing `println(“Hello world”);` to “Hello world” appearing in the terminal :)
